lib/spack/env/cc

```diff
@@ -108,6 +108,12 @@ case "$command" in
         ;;
     ld)
         mode=ld
+        for arg in "$@"; do
+            if [ "$arg" = -S ]; then
```

Comment:
Why is there an `as` in front of the `ld` in the output? That should not be there.